react/src/base/inputs/SprkInput/SprkInput.js

```diff
@@ -0,0 +1,140 @@
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import uniqueId from 'lodash/uniqueId';
+
+class SprkInput extends Component {
+  constructor(props) {
+    const { value } = props;
+    super(props);
+
+    if (value) {
+      this.state = {
+        hasValue: true,
+      };
+    } else {
+      this.state = {
+        hasValue: false,
+      };
+    }
+  }
+
+  render() {
+    const {
+      variant,
+      additionalClasses,
+      analyticsString,
+      formatter,
+      forwardedRef,
+      idString,
+      isDisabled,
+      isValid,
+      ariaDescribedBy,
+      value,
+      id,
+      ...rest
+    } = this.props;
+    const { hasValue } = this.state;
+
+    // Adds class for IE and Edge
+    const handleOnBlur = (e) => {
+      if (e.target.value.length > 0) {
+        this.setState({ hasValue: true });
+      } else {
+        this.setState({ hasValue: false });
```

Comment:
We'll need to check that they can run onBlur functions. We might have to add an onblur prop here.

Reply:
I don't see the added onBlur prop. Did you check this?